libfwupdplugin/fu-fmap-firmware.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2020 Benson Leung <bleung@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include "fu-firmware.h"
+
+#define FMAP_STRLEN		32	/* maximum length for strings, */
+					/* including null-terminator */
+
+#define FU_TYPE_FMAP_FIRMWARE (fu_fmap_firmware_get_type ())
+G_DECLARE_DERIVABLE_TYPE (FuFmapFirmware, fu_fmap_firmware, FU, FMAP_FIRMWARE, FuFirmware)
+
+struct _FuFmapFirmwareClass
+{
+	FuFirmwareClass		 parent_class;
+	gboolean		 (*parse)		(FuFirmware	*self,
+							 GBytes		*fw,
+							 guint64	 addr_start,
+							 guint64	 addr_end,
+							 FwupdInstallFlags flags,
+							 GError		**error);
+};
```

Comment:
could you add:

    /*< private >*/
    gpointer	padding[14];

...here please -- then we can add more vfuncs in the future without increasing the size of the struct and breaking ABI.

Reply:
Sounds good. Will do.